src/native/crt.c

```diff
@@ -302,3 +302,15 @@ jlong JNICALL Java_software_amazon_awssdk_crt_CRT_awsNativeMemory(JNIEnv *env, j
     }
     return allocated;
 }
+
+jstring aws_jni_string_from_cursor(JNIEnv *env, const struct aws_byte_cursor *native_data) {
+    struct aws_string *string = aws_string_new_from_array(aws_jni_get_allocator(), native_data->ptr, native_data->len);
+    if (string == NULL) {
+        return NULL;
+    }
+
+    jstring java_string = (*env)->NewStringUTF(env, (const char *)string->bytes);
```

Comment:
we have a function for this now
aws_string_c_str

Reply:
Since it's a cursor, there's no guarantee that it's null-terminated.

Reply:
NM, I understand now